# Homework Introduction to Python

$\textbf{Ubah nama di bawah yaa!}$

- Nama: Agustina Sri Wardani
- IPK : umm

In [ ]:
import numpy as np

# Normal

## 1. Operasi Variabel

> Misal kita memiliki toko sepatu. Modal membeli 1 kodi sepatu adalah Rp 1.000.000. Setiap sepatu dijual seharga Rp 80.000. Buatlah variabel modal, kuantitas, dan harga_jual sesuai informasi di atas. Hitung profit yang didapatkan jika semua sepatu berhasil terjual, dan masukkan ke variabel profit.


In [ ]:
#1 kodi = 20 pcs
modal = 1000000 #modal 1 kodi
kuantitas = 20
harga_jual = 80000

profit = ((harga_jual * kuantitas) - modal)
print(profit)

600000


Jadi, dari profit di atas, dapat disimpulkan bahwa toko sepatu memperoleh untung sebesar 600.000 jika semua sepatu berhasil terjual

## 2. Package Python dan Dictionary

> Kita akan coba mengakses data scientific article dari PLOS corpus
    Menggunakan API request dengan kode berikut : 
```
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')
```
**perhatikan tanda kutipnya tidak salah ketika meng-copy ke dalam python

Pertanyaannya : 
Convert response.content dari request tersebut dari JSON Format ke dalam Dictionary, dan print-kan key di dalamnya apa saja




In [ ]:
import requests
response = requests.get('http://api.plos.org/search?q=title:"Drosophila" AND body:"RNA"&fl=id,abstract')

import json
dataset = json.loads(response.content)

dataset.keys()

dict_keys(['response'])

Jadi, dari url link di atas, kami menyimpulkan bahwa memiliki keys **response**

# Intermediate

## 3. Operasi dalam List

> Buatlah list dengan 10 nama kota di indonesia, kemudian:
1. Urutkan nama kota di dalam list berdasarkan abjad (a)
2. Ambil 4 kota pertama dari list tersebut (yang telah diurutkan) (b)
3. Ambil 3 kota terakhir dari list tersebut (yang telah diurutkan) (c)



In [ ]:
# daftar 10 kota

kota = ['Batam', 'Tanjungpinang', 'Tanjung Uban', 'Bintan', 'Moro', 'Karimun', 'Natuna', 'Tanjung Batu', 'Lingga', 'Galang']

kota_disusun = sorted(kota) #disusun berdasarkan abjad
print(kota_disusun)

kota_disusun1 = kota_disusun[0:4] #4 kota pertama dari list kota
print(kota_disusun1)

kota_disusun2 = kota_disusun[7:] #3 kota terakhir dari list kta
print(kota_disusun2)

['Batam', 'Bintan', 'Galang', 'Karimun', 'Lingga', 'Moro', 'Natuna', 'Tanjung Batu', 'Tanjung Uban', 'Tanjungpinang']
['Batam', 'Bintan', 'Galang', 'Karimun']
['Tanjung Batu', 'Tanjung Uban', 'Tanjungpinang']


Jadi, dari operasi list di atas, dapat disimpulkan bahwa 10 kota/kabupaten/pulau di Kepulauan Riau diantaranya Batam, Bintan, Galang, Karimun, Lingga, Moro, Natuna, Tanjung Batu, Tanjung Uban, Tanjungpinang

## 4. Function dalam List

> Team sales memiliki list total transaksi dari 10 merchant pada bulan Juni 2020, yaitu: 
```
total_transaksi = [ 19, 35, 23, 25, 10, 15, 33, 45, 21, 30 ]
```
1. Trx minimum
2. Rata-rata 
3. Trx maksimal



In [ ]:
total_transaksi = [19, 35, 23, 25, 10, 15, 33, 45, 21, 30]

#transaksi minimal
trx_min = min(total_transaksi)
print(trx_min)

#rata-rata transaksi
rata2 = sum(total_transaksi) / len(total_transaksi)
print(rata2)

#transaksi maksimal
trx_max = max(total_transaksi)
print(trx_max)

10
25.6
45


Jadi, dari hasil operasi list di atas, dapat disimpulkan bahwa nilai minimal transaksi dari 10 merchant adalah 10, nilai maksimal adalah 45, dan rata-ratanya adalah 25.6

# Advanced

## 5. NumPy Array

> Buatlah array dari 5 bilangan ganjil pertama (dimulai dari 1), Kemudian lakukan perintah berikut : 
1. Kali-kan array tersebut dengan 10
2. Bagi-kan array (hasil dari bagian a) dengan 3
3. Kali-kan array (hasil dari bagian b) dengan array baru :
`[-45, 43, 3, 56, 5]`
4. Carilah pada index ke-berapa nilai tertinggi dan terendah dari array (hasil dari bagian c) 




In [ ]:
import numpy as np
array1 = np.array([1.234567, 2.345678])
print(np.round_(array1, decimals=2))

[1.23 2.35]


In [ ]:
import numpy as np
ganjil = [1, 3, 5, 7, 9] #5 bilangan ganjil pertama dimulai dari dari 1
bilangan_ganjil = np.array(ganjil)
print(np.round_(bilangan_ganjil, decimals=2))

#mengalikan array dengan 10
kali = bilangan_ganjil * 10
print(kali)

#membagi array dengan 3
bagi = bilangan_ganjil / 3
print(bagi)

#mengalikan hasil bagi b dengan array [-45, 43, 3, 56, 5]
new_array = [-45, 43, 3, 56, 5]
kalikan = bagi * new_array
print(kalikan)

#index nilai tertinggi dan terandah
index_maksimal = np.argmax(kalikan)
print(index_maksimal)

index_minimal = np.argmin(kalikan)
print(index_minimal)




[1 3 5 7 9]
[10 30 50 70 90]
[0.33333333 1.         1.66666667 2.33333333 3.        ]
[-15.          43.           5.         130.66666667  15.        ]
3
0


Jadi, dari hasil operasi numpy di atas, dapat disimpulkan bahwa 5 bilangan ganjil pertama adalah 1,3,5,7,9

# Terimakasih, salam dari Batam